hbase-balancer/src/test/java/org/apache/hadoop/hbase/master/balancer/TestStochasticLoadBalancer.java

```diff
@@ -291,14 +293,15 @@ public void testUpdateStochasticCosts() {
     } finally {
       //reset config
       conf.setFloat("hbase.master.balancer.stochastic.minCostNeedBalance", minCost);
-      conf.unset(HConstants.HBASE_MASTER_LOADBALANCE_BYTABLE);
+      conf.setBoolean(HConstants.HBASE_MASTER_LOADBALANCE_BYTABLE, oldIsByTable);
```

Comment:
@d-c-manning  Instead of resetting the conf object  in every method's finally block, can we reset the conf object to some default values in a method which is annotated `After` ?

Reply:
@shahrs87 I can try that. I would probably try it in the base class with a `@Before` annotation to ensure that we restore state before each test.

Reply:
It would require #4384 but since that is merged now, I guess we can try.

Reply:
I couldn't put it in the base class, because other subclasses override the `beforeAllTests` with their own `BeforeClass` version. It's probably for the best anyway, to keep the scope of the change more limited.

Reply:
@shahrs87 I made the change. It's best viewed by using the "hide whitespace" feature in github, because of the removal of `try` blocks. I think it cleans up this test class nicely, even though it's a bigger scoped change. Thanks for the recommendation!